### PR TITLE
ci: Provide external juju controller support

### PR DESCRIPTION
**Pull request by jneo8**
- Support runs-on option for self-hosted runner
- Option to skip action-operator action
- Secrets and steps to setup exteranl juju controller connection for functional testing
- Default to use organization's runner and action-operator.

### CODE REVIEW

**Review by Pjack — CHANGES_REQUESTED** [review]
some questions about the default value. It seems some default value is chosen for self-host, but some default value is chosen for IS runner.

File: .github/workflows/func.yaml [hunk 1]
```diff
@@ -49,11 +49,48 @@ on:
         required: false
         type: string
         default: "2.9/stable"
+      # runs-on for self-hosted runner
+      # workaround from: https://github.com/orgs/community/discussions/11692#discussioncomment-3541856
+      runs-on:
+        required: false
+        type: string
+        default: "['self-hosted']"
```
[the remark this answers is not in the document]
Comment (jneo8) [reply .github/workflows/func.yaml:57]: If using github official runner is default, we can change it. But I through we are using our organization's runner as default, then `self-hosted` is needed.

File: .github/workflows/func.yaml [hunk 2]
```diff
@@ -49,11 +49,48 @@ on:
...
+        description: "JSON format list of labels for runs-on"
+      action-operator:
+        required: false
```
[the remark this answers is not in the document]
Comment (jneo8) [reply .github/workflows/func.yaml:60]: The default value is true.

File: .github/workflows/func.yaml [hunk 3]
```diff
@@ -49,11 +49,48 @@ on:
...
+        type: boolean
+        default: true
+        description: "Use action-operator to setup environment or not"
+      external-controller:
+        required: false
+        type: boolean
+        default: false
```
[the remark this answers is not in the document]
Comment (jneo8) [reply .github/workflows/func.yaml:67]: Yes.

**Review by Pjack — APPROVED** [review]
No questions about the default value. It's good to me since it is verified in serverstack PoC environment.

**Pull request merged**